pkg/operator/replacement_starter.go

```diff
@@ -0,0 +1,323 @@
+package operator
+
+import (
+	"context"
+	"errors"
+	"os"
+	"reflect"
+	"time"
+
+	"k8s.io/client-go/dynamic/dynamicinformer"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	operatorclient "github.com/openshift/client-go/operator/clientset/versioned"
+	operatorinformer "github.com/openshift/client-go/operator/informers/externalversions"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	libgoetcd "github.com/openshift/library-go/pkg/operator/configobserver/etcd"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"github.com/openshift/multi-operator-manager/pkg/library/libraryapplyconfiguration"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/clock"
+)
+
+type AuthenticationOperatorInput struct {
```

Comment:
why not to make it private?

Reply:
doing this in https://github.com/openshift/multi-operator-manager/pull/12 and will revendor and update here